packages/playwright/src/runner/testServer.ts

```diff
@@ -182,18 +198,42 @@ class Dispatcher {
     await run;
   }
 
+  async findRelatedTestFiles(params:  {
+    configFile: string;
+    files: string[];
+  }): Promise<FindRelatedTestFilesReport> {
+    const config = await this._loadConfig(params.configFile);
+    const runner = new Runner(config);
+    return runner.findRelatedTestFiles('out-of-process', params.files);
+  }
+
+  async stop(params: {
+    configFile: string;
+  }) {
+    await this._stopTests();
+  }
+
+  async closeGracefully() {
+    gracefullyProcessExitDoNotHang(0);
+  }
+
   private async _stopTests() {
     this._testRun?.stop?.resolve();
     await this._testRun?.run;
   }
 
-  private async _loadConfig(overrides: ConfigCLIOverrides) {
-    return await loadConfig(this._configLocation, overrides);
-  }
-
   private _dispatchEvent(method: string, params: any) {
     this._ws.send(JSON.stringify({ method, params }));
   }
+
+  private async _loadConfig(configFile: string, overrides?: ConfigCLIOverrides): Promise<FullConfigInternal> {
+    return loadConfig({ resolvedConfigFile: configFile, configDir: path.dirname(configFile) }, overrides);
+  }
+
+  private _syncEnv(env: NodeJS.ProcessEnv) {
+    for (const name in env)
+      process.env[name] = env[name];
```

Comment:
I still think we might get bugs because of not restoring `env` after commands. What do we even pass as `env` here?

Reply:
We are passing reporter parameters here, it is a temporary measure.